python/src/FieldToPointFunctionalChaosAlgorithm_doc.i.in

```diff
@@ -0,0 +1,318 @@
+      \begin{array}{ccccl}
+         \cM_N \times (\Rset^d)^N & \rightarrow & \Rset^{K_T} & \rightarrow & \Rset^p \\
+         \mat{X} & \mapsto & \vect{Q} & \mapsto & \vect{Y}
+      \end{array}
+    \right.
+
+A limitation of this approach is that the projected modes sample has a dimension
+:math:`K_T` so the dimension of the input fields :math:`\mat{X_i}`
+and the associated number of modes must remain modest (curse of dimensionality).
+
+Parameters
+----------
+x : :class:`~openturns.ProcessSample`
+    Input process sample.
+y : :class:`~openturns.Sample`
+    Output sample.
+
+See also
+--------
+FieldFunctionalChaosSobolIndices, KarhunenLoeveAlgorithm, FunctionalChaosAlgorithm
+
+Notes
+-----
+As the input process decomposition is done with the values decomposition approach,
+it is possible to control the number of modes retained per input,
+the idea being to avoid a large input dimension for the chaos decomposition step.
+As done in :class:`~openturns.KarhunenLoeveSVDAlgorithm`, the :func:`setThreshold`
+and :func:`setNbModes` allow to control the spectrum ratio and maximum count.
+
+By default the input process data is assumed to not be centered, and the method
+:func:`setCenteredSample` allows to skip centering of the data for both the
+computation of modes coefficients and inside the metamodel.
+
+In the case of homogenous data (if variables have the same unit or scale),
+it is also possible to recompress the modes at the global level with :func:`setRecompress`.
+When enabled, the eigenvalues are gathered and sorted so as to find a global
+spectrum cut-off value by which the spectrum of each input are truncated.
+The default value can be set through the :class:`~openturns.ResourceMap` key
+*FieldToPointFunctionalChaosAlgorithm-DefaultRecompress*.
+
+For the chaos metamodel step, it is possible to specify the basis size with
+the :class:`~openturns.ResourceMap` key *FunctionalChaosAlgorithm-BasisSize*.
+
+It is possible to group input variables by independent blocks with :func:`setBlockIndices`,
+this way Karhunen-Loeve process decompositions are done on each block rather
+than on each variable thus reducing the total number of modes and help reduce
+the chaos input dimension.
+
+Examples
+--------
+>>> import openturns as ot
+>>> ot.RandomGenerator.SetSeed(0)
+>>> mesh = ot.RegularGrid(0.0, 0.1, 20)
+>>> cov = ot.KroneckerCovarianceModel(ot.MaternModel([2.0], 1.5), ot.CovarianceMatrix(4))
+>>> X = ot.GaussianProcess(cov, mesh)
+>>> x = X.getSample(500)
+>>> y = [[m[0] + m[1] + m[2] - m[3] + m[0] * m[1] - m[2] * m[3] - 0.1 * m[0] * m[1] * m[2]] for m in [xi.computeMean() for xi in x]]
+>>> algo = ot.FieldToPointFunctionalChaosAlgorithm(x, y)
+>>> algo.setThreshold(4e-2)
+>>> bs = ot.ResourceMap.GetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize')
+>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', 100)
+>>> algo.run()
+>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', bs)
```

Comment:
This business with the resource map key should be explained as a comment in the code, I think. Otherwise users might wonder why you are setting something only to unset it 2 lines below.

```suggestion
>>> # Temporarily lower the basis size for the sake of this example.
>>> # We need to store the original size.
>>> bs = ot.ResourceMap.GetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize')
>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', 100)
>>> algo.run()
>>> # The algorithm has been run with the lower basis size:
>>> # we can now restore the original value.
>>> ot.ResourceMap.SetAsUnsignedInteger('FunctionalChaosAlgorithm-BasisSize', bs)
```